src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala

```diff
@@ -1852,5 +1852,62 @@ class AppsControllerTest extends UnitTest with GroupCreation with ScalatestRoute
         (appJson \ "fetch" \ 0 \ "destPath" get) should be(JsString("bash.copy"))
       }
     }
+
+    "Replace multiple existing applications" in new Fixture {
+      Given("An app and group")
+      val app1Id = "/app1"
+      val app2Id = "/app2"
+
+      val newApp1Cmd = "bla1"
+      val newApp2Cmd = "bla2"
+      val apps = Seq(App(
+        id = app1Id,
+        cmd = Some("cmd1")
+      ), App(
+        id = app2Id,
+        cmd = Some("cmd1")
+      )
+      )
+
+      val normed = apps.map(normalize)
+      val appDefs = normed.map(Raml.fromRaml(_))
+      val rootGroup = createRootGroup(
+        appDefs.map { appDef =>
+        appDef.id -> appDef
+      }.toMap
+      )
+      val plan = DeploymentPlan(rootGroup, rootGroup)
+      groupManager.updateRoot(any, any, any, any, any) answers { args =>
+        val fn = args(1).asInstanceOf[RootGroup => RootGroup]
+        val updated = fn(rootGroup)
+        updated.app(PathId(app1Id)).get.cmd.get shouldEqual newApp1Cmd
+        updated.app(PathId(app2Id)).get.cmd.get shouldEqual newApp2Cmd
+        Future.successful(plan)
+      }
+      groupManager.rootGroup() returns rootGroup
+      groupManager.app(appDefs(0).id) returns Some(appDefs(0))
+      groupManager.app(appDefs(1).id) returns Some(appDefs(1))
+
+      When("The application is updated")
+      val body =
+        s"""[
+          | {
+          |   "id": "$app1Id",
+          |   "cmd": "$newApp1Cmd"
+          | },
+          | {
+          |   "id": "$app2Id",
+          |   "cmd": "$newApp2Cmd"
+          | }
+          ]""".stripMargin.getBytes("UTF-8")
+
+      When("The application is updated")
+      val entity = HttpEntity(body).withContentType(ContentTypes.`application/json`)
+      Put(Uri./, entity) ~> route ~> check {
+        Then("The application is updated")
+        status shouldEqual StatusCodes.OK
+        header[Headers.`Marathon-Deployment-Id`] should not be 'empty
+      }
+    }
```

Comment:
I'd like to see another test case covering `AppNotFoundException`.

Reply:
Another test won't make much sense because this endpoint will always allow creation